microsoft/testsuites/performance/storageperf.py

```diff
@@ -187,6 +194,109 @@ def perf_storage_over_nfs_synthetic_tcp_4k(self, result: TestResult) -> None:
+        fio = node.tools[Fio]
+        rm = node.tools[Rm]
+        fio_result = {}
+
+        for testcase in testcases:
+            test_name = testcase["testcase_name"]
+            fio_result[test_name] = {}
+            fio_result[test_name]["runtime_param"] = testcase
+
+            try:
+                path = node.get_working_path()
+
+                log.info(f"Executing the FIO testcase : {test_name}")
+
+                name = testcase.get("name", None)
+                filename = testcase.get("filename", None)
+                mode = testcase.get("mode", None)
+                iodepth = testcase.get("iodepth", None)
+                numjob = testcase.get("numjob", None)
+                time = testcase.get("time", 120)
+                ssh_timeout = testcase.get("ssh_timeout", 6400)
+                block_size = testcase.get("block_size", "4k")
+                size_mb = testcase.get("size_mb", 0)
+                direct = testcase.get("direct", True)
+                gtod_reduce = testcase.get("gtod_reduce", False)
+                ioengine = testcase.get("ioengine", "libaio")
+                group_reporting = testcase.get("group_reporting", True)
+                overwrite = testcase.get("overwrite", False)
+                time_based = testcase.get("time_based", False)
+
+                result = fio.launch(
+                    name=name,
+                    filename=filename,
+                    mode=mode,
+                    iodepth=iodepth,
+                    numjob=numjob,
+                    time=time,
+                    ssh_timeout=ssh_timeout,
+                    block_size=block_size,
+                    size_gb=size_mb,
+                    direct=direct,
+                    gtod_reduce=gtod_reduce,
+                    ioengine=ioengine,
+                    group_reporting=group_reporting,
+                    overwrite=overwrite,
+                    time_based=time_based,
+                    cwd=path
+                )
+
+                fio_result[test_name]["status"] = "Succeeded"
+                fio_result[test_name]["error"] = ""
+                fio_result[test_name]["metrics"] = {
+                    "qdepth": result.qdepth,
+                    "mode": result.mode,
+                    "iops": str(result.iops),
+                    "latency": str(result.latency),
+                    "iodepth": result.iodepth,
+                }
+
+            except Exception as e:
+                fio_result[test_name]["status"] = "Failed"
+                fio_result[test_name]["error"] = str(e)
+                fio_result[test_name]["metrics"] = {}
+                log.error(e)
+
+            rm.remove_directory(str(path.joinpath(f"{filename}*")), sudo=True)
+
+        log.debug(fio_result)
+
+        # Saving the result
+        report_path = node.get_working_path().joinpath("fio_report")
+        report_file = report_path.joinpath("fio_test_result.json")
```

Comment:
Don't use a file to dump the test results, it's not extendable. Use standard way, send sub tests messages to dump the results.

Reply:
Sure Chi. I will do it.

Reply:
I just realized it's another fio test. Please work with Lili to share code as much as possible.

Reply:
leveraging use of existing perf_disk function